src/manifest.c

```diff
@@ -616,16 +616,27 @@ struct list *create_update_list(struct manifest *current, struct manifest *serve
 		file = list->data;
 		list = list->next;
 
-		/* Skip files that aren't updated */
-		fullname = mk_full_filename(path_prefix, file->filename);
-		if (fullname == NULL) {
-			abort();
-		}
-		if (verify_file(file, fullname)) {
+		if (file->peer) {
+			/* Skip files that aren't updated, but only if they are
+			 * installed in the current version (tracked). */
+			if (file->is_tracked &&
+			    hash_equal(file->hash, file->peer->hash)) {
```

Comment:
After revisiting this, I don't think the "tracked" status (or deleted status) of a file matters for this check after all.

It should be sufficient to check for the peer and matching hashes before running verify_file().